test/test_parse_shebang.py

```diff
@@ -146,3 +148,27 @@ def test_throw_runtime_error_when_weird_object_passed(self):
         """Raise RuntimeError when weird object is passed in."""
         with ExpectedException(RuntimeError):
             parse(object())
+
+    # suppress(no-self-use)
+    def test_unicode(self):
+        """A filename can be text which in python 2 is not `str`."""
+        tmpdir = mkdtemp()
+        self.addCleanup(shutil.rmtree, tmpdir)
+
+        filename = os.path.join(tmpdir, u"foo.sh")
+        with io.open(filename, "w") as f:
```

Comment:
This is really just a matter of curiosity - what is the difference between using the `io` module here and just using `open`? I think this is the first time I've actually come across the module (forgive me, the python standard library is huge).

Having a quick skim over the documentation, it looks like the module brings whatever IO semantics python 3 had to python 2 in an opt-in manner. Is that right?